elastic/kibana
# [Ingest Pipelines] Stateful uls for search and filters in pipelines table

## Summary

Editing pipelines that belong to the same search can be annoying as the user needs to:

* Search/filter
* Then select the pipeline
* Go to Edit
* Do your changes, save, go back
* Your search filter is lost and need to filter/search again

This PR addresses this problem by persisting the filters and search as query params in the url.

##### How to test

* Start up kibana and es
* Navigate to Stack management -> Ingest pipelines
* Verify that:
  * When searching or filtering the filters/search are persisted in URL as query params. Hard reloading this page should initialize the filters/search with the values of the query params.
  * When having search or filters in query params, open the details flyout and check that the pipeline is now also included in the url
  * When closing the pipeline details flyout, should send the user back to the list of pipelines and should keep whatever query params it previously had.
